virttest/utils_test/libvirt.py

```diff
@@ -1786,15 +1787,24 @@ def create_controller_xml(contr_dict, oper="get_xml", vm_name=None):
     contr_index = contr_dict.get("controller_index")
     contr_alias = contr_dict.get("contr_alias")
     contr_addr = contr_dict.get("controller_addr")
+    contr_busNr = contr_dict.get("controller_busNr")
+    contr_node = contr_dict.get("controller_node")
     contr = controller.Controller(contr_type)
     contr.model = contr_model
     if contr_index:
         contr.index = contr_index
     if contr_alias:
         contr.alias = dict(name=contr_alias)
+    if contr_busNr:
+        contr.target = dict(busNr=contr_busNr)
+    if contr_node:
+        contr.node = contr_node
     if contr_addr:
         contr.address = contr.new_controller_address(attrs=eval(contr_addr))
     if oper == "add_controller":
+        if vmxml_ori:
```

Comment:
previously this method has existed parameter vm_name.  Can vmxml_ori be provided by vm_name itself ,rather than add additional parameter?

Reply:
@chunfuwen , if use 'vm_name', then we have to dump vm to get guest xml, but that is not what I need. Sometimes, I need a fresh vm xml customized by myself here which I create a new controller in it.